### DISCUSSION TITLE
Question about `Selection` `include` usage

### DISCUSSION THREAD

**Discussion by kolmakova** [uwdata/mosaic#570] [Q&A] [answered]
Hi! Noticed a new `include` option for `Selection` in the update v0.11, and I have a question about how it works.

I'm using Mosaic Core with the `Selection.crossfilter`. I have several custom interactive components that publish clauses into this selection. I want to have the ability to disable crossfiltering for some components but still include them in the `Selection` (have ability to enable it later). Specifically, interactions that are not crossfiltered should return results unaffected by the current clauses from other components. 

What is the correct way to exclude a component from `Selection` crossfiltering? Should I create a `Selection.single` for each interactive component and use `include` to include them in the `Selection.crossfilter` only when I want to crossfilter those components? Is it possible to update `include` without recreating `Selection`?

**Comment by jheer** [comment, marked as answer]
If possible, an ideal setup is to (1) first define selections at the finest granularity you need, (2) create downstream selections that use `include` to incorporate clauses from those upstream selections.

However, if you want to *dynamically* change inclusion/exclusion of clauses within a selection, you'll need to go beyond this setup that Mosaic expects. While not currently part of the "official" selection API, you can change the inclusion relays in JavaScript. A basic selection (step 1 above) "relays" clauses to those that include it (step 2 above). These downstream selections are stored within the upstream selection's `Selection._relay` property, which is of type `Set<Selection>`. You can (at your own risk) try modifying this by adding/removing downstream selections to target. However, at the same time you'll need to make sure any existing clauses that have been already been relayed to downstream selections are also removed.

**Reply by kolmakova** [reply]
Thank you for the detailed explanation. I'll try implementing the approach you suggested. Really appreciate you taking the time to explain both the recommended pattern and the advanced option.

Mosaic is awesome! Wishing the team great success with its development! ☺️

**Comment by caioross** [comment]
Lendo a v1.7.0 com os olhos de quem quer que as skills gerem receita — e desta vez tem uma correção que literalmente mexe com os meus números.

**O bug dos totais de ganhos: finalmente corrigido — e agora preciso saber o estrago**

A correção do #615 diz que antes só as 100 últimas transações eram somadas para calcular o total de ganhos. Se você tinha mais de 100 transações, via um número errado. Eu não sei se passei dessa marca — e não consigo checar facilmente. Mas o ponto é: isso era um bug *silencioso*. Não havia aviso nenhum de que o número estava errado, e quem dependia desse total para calcular quanto ia sacar (ou para entender se a skill estava performando) tomava decisões com dado estragado.

Celebro a correção — mas ficou uma pergunta: **existe alguma forma de ver os totais históricos corrigidos retroativamente no painel?** Quero comparar o que via antes com o que é real agora. Se o número mudou muito, preciso saber.

**O selo de segurança é bom para quem tem — e invisível para quem não tem**

O selo verde dá credibilidade na página da skill. Entendo que "ausente ≠ negativo" no design do produto — mas o comprador médio não leu essa nota de rodapé. Na prática, se minha skill ainda não foi avaliada e aparece sem nenhum badge ao lado de uma que tem o verde, a comparação visual me penaliza.

O que eu precisaria: **existe alguma fila de solicitação de avaliação?** Posso pedir que a minha skill entre no processo de *vetting* de segurança? Hoje parece que o critério de quem é avaliado primeiro é opaco para o criador.


> Feedback de usuário (persona **Yuki, a Criadora/Vendedora**, programa de personas do SkillDepot).
